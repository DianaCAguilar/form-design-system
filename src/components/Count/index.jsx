import React from 'react';
import PropTypes from 'prop-types';
import cc from 'classcat';

export const VALID_TYPES = ['notification', 'passive'];

const Count = ({ label, type = 'notification' }) => {
  const isNotification = type === 'notification';

  return (
    <div className="fds-countWrapper alignChild--center--center">
      <div
        className={cc([
          'fds-count',
          'padding--x--s fontSize--xs alignChild--center--center',
          {
            'shape--circle fds-count--dot color--white': !label,
            'color--white bgColor--red': isNotification,
            'color--aqua': !isNotification,
            'bgColor--aqua': !isNotification && !label,
          },
        ])}
      >
        <span className="fds-count-label wrap--singleLine--truncate">{label}</span>
      </div>
    </div>
  );
};

Count.propTypes = {
  /**
   * String to render inside the Count bubble.
   * The `element` type should be reserved for FDS icons.
   */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),

  /** Type of Count */
  type: PropTypes.oneOf(VALID_TYPES),
};

export default Count;
