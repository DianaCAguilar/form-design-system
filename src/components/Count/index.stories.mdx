import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import { withKnobs, text, optionsKnob as options } from '@storybook/addon-knobs';
import ImportPath from 'util/storybook-docs/ImportPath';
import { arrayToOptions } from 'util/storybook-docs/knobHelpers';

import Flex from 'components/Flex';
import FlexItem from 'components/FlexItem';
import GapList from 'components/GapList';

import Count, { VALID_TYPES } from '.';

<Meta title="Components/Count" decorators={[withKnobs]} component={Count} />

# Count

<ImportPath component={Count}  />

Renders a count bubble with an optional label. The bubble will be centered within the
parent container you place it in.

To render a `Count` in the top right corner of an element, use the `Updatable` component.

<Story name="Knobs" parameters={{ docs: { disable: true }}}>
  <Count
    label={text('label', '42')}
    type={options('type', arrayToOptions(VALID_TYPES), 'notification', {
      display: 'inline-radio',
    })}
  />
</Story>

<Preview isExpanded>
  <Count
    label="42"
  />
</Preview>

<Props of={Count} />

### Variations

Count uses the `notification` type by default, which is used to indicate a count of
_new_ items behind a link.

The `passive` type can be used to show total items.

When the `label` is left undefined, both types will collapse into a dot, indicating there
is at least one item.

<Preview>
  <Story name="Variations">
    <GapList
      items={[
        <Count label="42" />,
        <Count label="42" type="passive" />,
        <Count />,
        <Count type="passive" />,
      ]}
    />
  </Story>
</Preview>

### Table rows with Count

To indicate a table row is "new", place the indicator to the left of the first column.
The bubble will center itself within the nearest parent.

<Preview>
  <Story name="With table rows">
    <Flex>
      <FlexItem shrink>
        <div style={{width: '100%', height: '32px'}}>
          <Count />
        </div>
      </FlexItem>
      <FlexItem>
        <div className="bgColor--haze" style={{width: '100%', height: '32px'}} />
      </FlexItem>
    </Flex>
  </Story>
</Preview>
