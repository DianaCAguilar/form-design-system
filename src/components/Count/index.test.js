import React from 'react';
import { shallow } from 'enzyme';

import Count from '.';

describe('Count component', () => {

  it('matches snapshot (default props, has label)', () => {
    const component = shallow(<Count label="test" />);
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot (update dot)', () => {
    const component = shallow(<Count />)
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot (count dot)', () => {
    const component = shallow(<Count type="passive" />)
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot (count with label)', () => {
    const component = shallow(<Count label="42" type="passive" />)
    expect(component).toMatchSnapshot();
  });
});
