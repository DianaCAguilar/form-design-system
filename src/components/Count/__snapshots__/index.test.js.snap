// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component matches snapshot (count dot) 1`] = `
<div
  className="fds-statusWrapper alignChild--center--center"
>
  <div
    className="fds-status padding--x--s fontSize--xs alignChild--center--center shape--circle fds-status--dot color--white color--aqua bgColor--aqua"
  >
    <span
      className="fds-status-label wrap--singleLine--truncate"
    />
  </div>
</div>
`;

exports[`Status component matches snapshot (count with label) 1`] = `
<div
  className="fds-statusWrapper alignChild--center--center"
>
  <div
    className="fds-status padding--x--s fontSize--xs alignChild--center--center color--aqua"
  >
    <span
      className="fds-status-label wrap--singleLine--truncate"
    >
      42
    </span>
  </div>
</div>
`;

exports[`Status component matches snapshot (default props, has label) 1`] = `
<div
  className="fds-statusWrapper alignChild--center--center"
>
  <div
    className="fds-status padding--x--s fontSize--xs alignChild--center--center color--white bgColor--red"
  >
    <span
      className="fds-status-label wrap--singleLine--truncate"
    >
      test
    </span>
  </div>
</div>
`;

exports[`Status component matches snapshot (update dot) 1`] = `
<div
  className="fds-statusWrapper alignChild--center--center"
>
  <div
    className="fds-status padding--x--s fontSize--xs alignChild--center--center shape--circle fds-status--dot color--white color--white bgColor--red"
  >
    <span
      className="fds-status-label wrap--singleLine--truncate"
    />
  </div>
</div>
`;
